agreement_legal/models/agreement.py

```diff
@@ -465,44 +461,22 @@ def copy(self, default=None):
     def _exclude_readonly_field(self):
         return ["stage_id"]
 
+    def _get_agreement_readonly_domain(self):
+        return "bool(readonly)"
+
     @api.model
-    def fields_view_get(
-        self, view_id=None, view_type=False, toolbar=False, submenu=False
-    ):
-        res = super().fields_view_get(
-            view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu
-        )
+    def get_view(self, view_id=None, view_type=False, **options):
+        res = super().get_view(view_id, view_type, **options)
         # Readonly fields
         if view_type == "form":
             doc = etree.XML(res["arch"])
-            for node in doc.xpath("//field"):
+            for node in doc.xpath("//field[@name][not(ancestor::field)]"):
                 if node.attrib.get("name") in self._exclude_readonly_field():
                     continue
-                attrs = ast.literal_eval(node.attrib.get("attrs", "{}"))
-                if attrs:
-                    if attrs.get("readonly"):
-                        attrs["readonly"] = ["|", ("readonly", "=", True)] + attrs[
-                            "readonly"
-                        ]
-                    else:
-                        attrs["readonly"] = [("readonly", "=", True)]
-                else:
-                    attrs["readonly"] = [("readonly", "=", True)]
-                node.set("attrs", simplejson.dumps(attrs))
-                modifiers = ast.literal_eval(
-                    node.attrib.get("modifiers", "{}")
-                    .replace("true", "True")
-                    .replace("false", "False")
-                )
-                readonly = modifiers.get("readonly")
-                invisible = modifiers.get("invisible")
-                required = modifiers.get("required")
-                if isinstance(readonly, bool) and readonly:
-                    attrs["readonly"] = readonly
-                if isinstance(invisible, bool) and invisible:
-                    attrs["invisible"] = invisible
-                if isinstance(required, bool) and required:
-                    attrs["required"] = required
-                node.set("modifiers", simplejson.dumps(attrs))
+                new_r_modifier = self._get_agreement_readonly_domain()
+                old_r_modifier = node.attrib.get("readonly")
+                if old_r_modifier:
+                    new_r_modifier = f"({old_r_modifier}) or ({new_r_modifier})"
```

Comment:
This is causing some issues in views where the `readonly` field is not added.
E.g.: https://github.com/OCA/field-service/actions/runs/11960339984/job/33344121836?pr=1267

```
2024-11-21 19:35:27,928 519 ERROR odoo odoo.addons.fieldservice_agreement.tests.test_fsm_agreement: ERROR: FSMOrder.test_fsm_agreement
Traceback (most recent call last):
  File "/opt/odoo/odoo/tools/safe_eval.py", line 391, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
NameError: name 'readonly' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/fieldservice_agreement/tests/test_fsm_agreement.py", line 31, in test_fsm_agreement
    f.name = "Test Agreement"
  File "/opt/odoo/odoo/tests/form.py", line 326, in __setattr__
    self[field_name] = value
  File "/opt/odoo/odoo/tests/form.py", line 333, in __setitem__
    assert not self._get_modifier(field_name, 'readonly'), f"can't write on readonly field {field_name!r}"
  File "/opt/odoo/odoo/tests/form.py", line 361, in _get_modifier
    return bool(safe_eval(expr, eval_context))
  File "/opt/odoo/odoo/tools/safe_eval.py", line 405, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'NameError'>: "name 'readonly' is not defined" while evaluating
'bool(readonly)'
```

To reproduce: try opening the original agreement form view

Reply:
@ivantodorovich I'm trying to reproduce on the UI but no success for the moment :/

Reply:
`2024-11-28 17:50:45,006 1 INFO testdb odoo.tests.stats: agreement_legal: 40 tests 1.06s 1037 queries 
2024-11-28 17:50:45,006 1 INFO testdb odoo.tests.stats: fieldservice_agreement: 3 tests 0.12s 84 queries 
2024-11-28 17:50:45,006 1 INFO testdb odoo.tests.result: 0 failed, 0 error(s) of 27 tests when loading database 'testdb' 
`